File: src/78.subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums)
    {
        List<IList<int>> result = new List<IList<int>>();
        int size = nums.Length;
        int totalN = (int)Math.Pow(2, size);
        int tmp = 0;
        for(int i = 0; i < totalN; ++i)
        {
            List<int> lst = new List<int>();
            for(int j = 0; j < size; ++j)
            {
                uint tmp2 = (uint)tmp;
                tmp2 >>= j;
                tmp2 <<= 31;
                tmp2 >>= 31;
                if (tmp2 == 1)
                    lst.Add(nums[j]);
            }
            result.Add(lst);
            tmp++;
        }
        return result;
    }
}

File: dump-cn/78.subsets.cpp
class Solution {
public:
	vector<vector<int>> subsets(vector<int>& nums) {
		vector<vector<int>> result;
		vector<int> tmp;
		dfsSubset(result, nums, tmp, 0);
		return result;
	}

	void dfsSubset(vector<vector<int>>& result, vector<int>& nums, vector<int>& cur, int index) {
		if (index == nums.size()) {
			result.push_back(cur);
			return;
		}
		// Get
		cur.push_back(nums[index]);
		dfsSubset(result, nums, cur, index + 1);
		cur.pop_back();
		// Not Get
		dfsSubset(result, nums, cur, index + 1);
	}
};